dataplane/handlers/interface.go

```diff
@@ -0,0 +1,293 @@
+
+	tapName := engine.IntfNameToTapName(config.GetName())
+	state := ni.state[config.GetName()]
+
+	if config.GetOrCreateEthernet().MacAddress != nil {
+		if config.GetEthernet().GetMacAddress() != state.GetEthernet().GetMacAddress() {
+			log.Infof("setting interface %s hw-addr %q", engine.IntfNameToTapName(config.GetName()), config.GetEthernet().GetMacAddress())
+			if err := kernel.SetInterfaceHWAddr(engine.IntfNameToTapName(config.GetName()), config.GetEthernet().GetMacAddress()); err != nil {
+				log.Warningf("Failed to set mac address of port: %v", err)
+				return
+			}
+		}
+	}
+	if config.GetOrCreateSubinterface(0).Enabled != nil {
+		if state.GetSubinterface(0).Enabled == nil || config.GetSubinterface(0).GetEnabled() != state.GetSubinterface(0).GetEnabled() {
+			log.Infof("setting interface %s enabled %t", engine.IntfNameToTapName(config.GetName()), config.GetSubinterface(0).GetEnabled())
+			if err := kernel.SetInterfaceState(engine.IntfNameToTapName(config.GetName()), config.GetSubinterface(0).GetEnabled()); err != nil {
+				log.Warningf("Failed to set state address of port: %v", err)
+				return
+			}
+		}
+	}
+	// TODO: refactor this.
+	for _, addr := range config.GetOrCreateSubinterface(0).GetOrCreateIpv4().Address {
+		configIP := addr.Ip
+		configPL := addr.PrefixLength
+		var stateIP *string
+		var statePL *uint8
+		stateAddr := state.GetSubinterface(0).GetIpv4().GetAddress(addr.GetIp())
+		if stateAddr != nil {
+			stateIP = stateAddr.Ip
+			statePL = stateAddr.PrefixLength
+		}
+
+		if configIP != nil && configPL != nil && (stateIP == nil || *statePL != *configPL) {
+			log.Infof("Config IP: %v, Config PL: %v. State IP: %v, State PL: %v", addr.GetIp(), addr.GetPrefixLength(), stateAddr.GetIp(), stateAddr.GetPrefixLength())
+			log.Infof("setting interface %s ip %s/%d", tapName, *configIP, *configPL)
+			if err := kernel.SetInterfaceIP(engine.IntfNameToTapName(config.GetName()), *configIP, int(*configPL)); err != nil {
+				log.Warningf("Failed to set ip address of port: %v", err)
+				return
+			}
+		}
+	}
+	for _, addr := range config.GetOrCreateSubinterface(0).GetOrCreateIpv6().Address {
+		configIP := addr.Ip
+		configPL := addr.PrefixLength
+		var stateIP *string
+		var statePL *uint8
+		stateAddr := state.GetSubinterface(0).GetIpv6().GetAddress(addr.GetIp())
+		if stateAddr != nil {
+			stateIP = stateAddr.Ip
+			statePL = stateAddr.PrefixLength
+		}
+
+		if configIP != nil && configPL != nil && (stateIP == nil || *statePL != *configPL) {
+			log.Infof("Config IP: %v, Config PL: %v. State IP: %v, State PL: %v", addr.GetIp(), addr.GetPrefixLength(), stateAddr.GetIp(), stateAddr.GetPrefixLength())
+			log.Infof("setting interface %s ip %s/%d", tapName, *configIP, *configPL)
+			if err := kernel.SetInterfaceIP(engine.IntfNameToTapName(config.GetName()), *configIP, int(*configPL)); err != nil {
+				log.Warningf("Failed to set ip address of port: %v", err)
+				return
+			}
+		}
+	}
+	// TODO: delete IPs
+}
```

Comment:
After reconciliation, need to update state data to the datastore

Reply:
added todo